Engines/Wine/Plugins/Windows version/script.js

```diff
@@ -15,18 +15,26 @@ Wine.prototype.windowsVersion = function (version, servicePack) {
     }
 
     // set
-    var regeditFileContent =
-        "REGEDIT4\n" +
-        "\n" +
-        "[HKEY_CURRENT_USER\\Software\\Wine]\n" +
-        "\"Version\"=\"" + version + "\"\n";
-
-    if (servicePack) {
-        var servicePackNumber = servicePack.replace("sp", "");
-        regeditFileContent += "[HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows NT\\CurrentVersion]\n";
-        regeditFileContent += "\"CSDVersion\"=\"Service Pack "+ servicePackNumber +"\"\n";
-        regeditFileContent += "[HKEY_LOCAL_MACHINE\\System\\CurrentControlSet\\Control\\Windows]\n";
-        regeditFileContent += "\"CSDVersion\"=dword:00000"+servicePackNumber+"00\n";
+    var regeditFileContent;
+    if (version == null) {
+        regeditFileContent =
+            "REGEDIT4\n" +
+            "\n" +
```

Comment:
This part is the same, so you could actually change the code to:
```Javascript
regeditFileContent =
    "REGEDIT4\n" +
    "\n" +
    "[-HKEY_CURRENT_USER\\Software\\Wine]";

if (version) {
...
```

Reply:
No it is not if you look well

Reply:
Ah the "-" in front of HKEY is different. Didn't see that.

Still would prefer
```Javascript
if (version)
```
over
```Javascript
if (version == null)
```

Reply:
Not sure because if version is undefined, then, we switch back to the getter. The I think in this particular case, we really need version to be explicitely null

Reply:
I see. It's not so nice from an API perspective.